### PR TITLE
refactor: update `blas/ext/base/dnansum` to follow current project conventions

### LINKED ISSUE #1483: [RFC]: refactor `blas/ext/base/dnansum` to follow current project conventions
### Description

This RFC proposes refactoring [`@stdlib/blas/ext/base/dnansum`](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/blas/ext/base/dnansum) to follow current project conventions.

These conventions are outlined in more detail in https://github.com/stdlib-js/stdlib/issues/788. Namely, we want to migrate from C++ add-on interfaces to C add-on interfaces and then make various style and simplification changes.

Some example commits which perform most, if not all, of the requested changes include https://github.com/stdlib-js/stdlib/commit/d66d80cfcbab1da95cb5ae8cfae4a1fc4c5129d1 (saxpy),  https://github.com/stdlib-js/stdlib/commit/3c82fe7b6c0f9fd6ef02c86d6e2024d708ac8c14 (cswap), and https://github.com/stdlib-js/stdlib/commit/5b15c9675be434c0e11b1ec1e1f78d7b089abbcd (dasum).

### Related Issues

Related issues #788, #1152.

### Questions

No.

### Other

No.

### Checklist

- [X] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [X] Searched for existing issues and pull requests.
- [X] The issue name begins with `RFC:`.

### PR DESCRIPTION
**Pull request by AuenKr**
Resolves #1483.

## Description

> What is the purpose of this pull request?

This pull request:

-   refactored `@stdlib/blas/ext/base/dnansum` to follow current project conventions

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves #1483 

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

File: lib/node_modules/@stdlib/blas/ext/base/dnansum/benchmark/benchmark.js [hunk 1]
```diff
@@ -39,18 +40,15 @@ var dnansum = require( './../lib/dnansum.js' );
...
 
-	x = new Float64Array( len );
-	for ( i = 0; i < x.length; i++ ) {
-		if ( randu() < 0.2 ) {
-			x[ i ] = NaN;
-		} else {
-			x[ i ] = ( randu()*10.0 ) - 20.0;
+	function clbk() {
+		if ( bernoulli( 0.7 ) > 0 ) {
+			return discreteUniform( -20, 10 );
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/ext/base/dnansum/benchmark/benchmark.js]:
```suggestion
			return discreteUniform( -10, 10 );
```

File: lib/node_modules/@stdlib/blas/ext/base/dnansum/benchmark/benchmark.native.js [hunk 2]
```diff
@@ -48,18 +49,15 @@ var opts = {
...
 
-	x = new Float64Array( len );
-	for ( i = 0; i < x.length; i++ ) {
-		if ( randu() < 0.2 ) {
-			x[ i ] = NaN;
-		} else {
-			x[ i ] = ( randu()*10.0 ) - 20.0;
+	function clbk() {
+		if ( bernoulli( 0.7 ) > 0 ) {
+			return discreteUniform( -20, 10 );
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/ext/base/dnansum/benchmark/benchmark.native.js]:
```suggestion
			return discreteUniform( -10, 10 );
```

File: lib/node_modules/@stdlib/blas/ext/base/dnansum/benchmark/benchmark.ndarray.js [hunk 3]
```diff
@@ -39,18 +40,15 @@ var dnansum = require( './../lib/ndarray.js' );
...
 
-	x = new Float64Array( len );
-	for ( i = 0; i < x.length; i++ ) {
-		if ( randu() < 0.2 ) {
-			x[ i ] = NaN;
-		} else {
-			x[ i ] = ( randu()*10.0 ) - 20.0;
+	function clbk() {
+		if ( bernoulli( 0.7 ) > 0 ) {
+			return discreteUniform( -20, 10 );
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/ext/base/dnansum/benchmark/benchmark.ndarray.js]:
```suggestion
			return discreteUniform( -10, 10 );
```

File: lib/node_modules/@stdlib/blas/ext/base/dnansum/benchmark/benchmark.ndarray.native.js [hunk 4]
```diff
@@ -48,18 +49,15 @@ var opts = {
...
 
-	x = new Float64Array( len );
-	for ( i = 0; i < x.length; i++ ) {
-		if ( randu() < 0.2 ) {
-			x[ i ] = NaN;
-		} else {
-			x[ i ] = ( randu()*10.0 ) - 20.0;
+	function clbk() {
+		if ( bernoulli( 0.7 ) > 0 ) {
+			return discreteUniform( -20, 10 );
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/ext/base/dnansum/benchmark/benchmark.ndarray.native.js]:
```suggestion
			return discreteUniform( -10, 10 );
```

File: lib/node_modules/@stdlib/blas/ext/base/dnansum/lib/ndarray.native.js [hunk 5]
```diff
@@ -37,20 +38,17 @@ var addon = require( './dnansum.native.js' );
...
 */
 function dnansum( N, x, stride, offset ) {
 	var view;
-	if ( stride < 0 ) {
-		offset += (N-1) * stride;
-	}
-	view = new Float64Array( x.buffer, x.byteOffset+(x.BYTES_PER_ELEMENT*offset), x.length-offset ); // eslint-disable-line max-len
+
+	offset = minViewBufferIndex( N, stride, offset );
+	view = offsetView( x, offset );
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/ext/base/dnansum/lib/ndarray.native.js:51]:
```suggestion
	view = offsetView( x, offset );

```

File: lib/node_modules/@stdlib/blas/ext/base/dnansum/manifest.json [hunk 6]
```diff
@@ -35,8 +36,40 @@
...
+      ],
+      "include": [
+        "./include"
+      ],
+      "libraries": [
+        "-lm"
+      ],
+      "libpath": [],
+      "dependencies": []
+    }
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/ext/base/dnansum/manifest.json:73]: Incorrect indentation

**Review by Pranavchiku — COMMENTED** [review]
Minor change requested, rest all is good. Once addressed this will be good to merge.

File: lib/node_modules/@stdlib/blas/ext/base/dnansum/manifest.json (hunk 6 above)
Comment (AuenKr) [reply lib/node_modules/@stdlib/blas/ext/base/dnansum/manifest.json:73]: fixed indentation issue to this PR and all other similar PR #1864 #1867 #1888 #1889 #1895 .
I apologize for raising all these PRs simultaneously, I will not do the same in the future.

**Review by Pranavchiku — APPROVED** [review]
LGTM. Thanks @AuenKr !

**Comment by AuenKr** [comment]
> LGTM. Thanks @AuenKr !

Can you also review the other similar PR  #1864 #1867 #1888 #1889 #1895 .

**Pull request merged**